src/azure-cli/azure/cli/command_modules/storage/_params.py

```diff
@@ -1592,15 +1594,48 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('restored_name',
                    help='A new file share name to be restored. If not specified, deleted share name will be used.')
 
+    with self.argument_context('storage share create') as c:
+        c.extra('share_name', share_name_type, options_list=('--name', '-n'), required=True)
+        c.argument('fail_on_exist', help='Specify whether to throw an exception when the share exists. False by '
+                                         'default.')
+        c.argument('quota', type=int, help='Specifies the maximum size of the share, in gigabytes. Must be greater '
+                                           'than 0, and less than or equal to 5TB (5120).')
+
     with self.argument_context('storage share url') as c:
         c.argument('unc', action='store_true', help='Output UNC network path.')
         c.argument('protocol', arg_type=get_enum_type(['http', 'https'], 'https'), help='Protocol to use.')
 
     with self.argument_context('storage share list') as c:
         c.argument('num_results', arg_type=num_results_type)
+        c.extra('marker', help='An opaque continuation token. This value can be retrieved from the next_marker field '
+                               'of a previous generator object if num_results was specified and that generator has '
+                               'finished enumerating results. If specified, this generator will begin returning '
+                               'results from the point where the previous generator stopped.')
+        c.extra('timeout', timeout_type)
+        c.argument('include_snapshots', help='Specifies that share snapshots be returned in the response.')
+        c.argument('include_metadata', help='Specifies that share metadata be returned in the response.')
+        c.extra('prefix',
+                help='Filter the results to return only blobs whose name begins with the specified prefix.')
+        c.ignore('name_starts_with')
 
     with self.argument_context('storage share exists') as c:
+        c.extra('share_name', share_name_type, options_list=('--name', '-n'), required=True)
+        c.extra('snapshot', options_list=['--snapshot'], is_preview=True,
```

Comment:
```suggestion
        c.extra('snapshot', options_list=['--snapshot'],
```
Why marking `snapshot` argument preview?